nextcord/http.py

```diff
@@ -149,83 +471,234 @@ def __exit__(
+        increase this parameter.
+    time_offset: :class:`float`
+        Amount of seconds added to all ratelimit timers for lag compensation.
+
+        Due to latency and Discord servers not perfectly time synced, having no offset can cause 429's to occur even
+        with us following the reported X-RateLimit-Reset-After.
+
+        Increasing will protect from erroneous 429s but will slow bucket resets, lowering max theoretical speed.
+
+        Decreasing will hasten bucket resets and increase max theoretical speed but may cause 429s.
+    default_auth: Optional[:class:`str`]
+        Default string to use in the Authorization header if it's not manually provided.
+    assume_unsync_clock: :class:`bool`
+        Whether to assume the system clock is unsynced regarding rate limit handling.
+
+        If ``True``, rate limits will use a discovered static-ish time delay to figure out when buckets reset.
+        Generally results in longer-than-necessary inaccurate bucket reset times, especially with high-latency
+        connections, but is most consistent across a wide variety of hosts and applications.
+
+        If ``False``, rate limits will use a combination of server-given bucket reset timestamps compared with local
+        time to accurately reset when the bucket resets and discovered static-ish time delays to predict the next reset
+        when needed.
+        Generally results in more accurate bucket reset times, especially with high-latency connections, but relies
+        heavily on host time being accurate with Discords time.
+    proxy: Optional[:class:`str`]
+        The proxy url to connect to the Discord API with, if any.
+    proxy_auth: Optional[:class:`aiohttp.BasicAuth`]
+        The authentication to use in order to make a request to the proxy url. Not to be confused with ``default_auth``.
+    dispatch: :class:`Callable`
+        The dispatcher to use for rate limit events.
+    """
 
     def __init__(
         self,
         connector: Optional[aiohttp.BaseConnector] = None,
         *,
+        max_global_requests: int = 50,
+        time_offset: float = 0.0,
+        default_auth: Optional[str] = None,
+        assume_unsync_clock: bool = False,  # TODO: Think about renaming this?
         proxy: Optional[str] = None,
         proxy_auth: Optional[aiohttp.BasicAuth] = None,
-        loop: Optional[asyncio.AbstractEventLoop] = None,
-        unsync_clock: bool = True,
         dispatch: Callable,
     ) -> None:
-        self.loop: asyncio.AbstractEventLoop = asyncio.get_event_loop() if loop is None else loop
-        self.connector = connector
+        # TODO: Think about adding ratelimit_multiplier? Would reduce the internal RateLimit.limit by that
+        #  float (such as 0.7) and could allow people to run multiple NC bots/processes on the same token while avoiding
+        #  ratelimit issues. Could also help with replit-style scenarios.
```

Comment:
Note with this approach, it assumes each process makes the seem amount of requests towards a bucket. However the following conditions are present:
- Seperate processes are unlikely to contribute equally
- Based on my ratelimit understanding and multiple processes, outside of the global ratelimit a single process shouldn't be sharing ratelimits with another process due to how sharding currently works

While I love the feature idea, a simple number is likely a bit too naive to get things right nicely and the complexity required to do it nicely would be too large imo

Reply:
that's fair, I'll probably remove that TODO when I get home.